include/libvfio-user.h

```diff
@@ -454,11 +455,12 @@ vfu_irq_message(vfu_ctx_t *vfu_ctx, uint32_t subindex);
  * @len: size of memory to be mapped
  * @sg: array that receives the scatter/gather entries to be mapped
  * @max_sg: maximum number of elements in above array
- * @prot: protection as define in <sys/mman.h>
+ * @prot: protection as defined in <sys/mman.h>
  *
  * @returns the number of scatter/gather entries created on success, and on
  * failure:
- *  -1:         if the GPA address span is invalid, or
+ *  -1:         if the GPA address span is invalid or
```

Comment:
We need to clear `errno`  if the `if` is taken in L407, so that the caller can figure out what `-1` exactly means. We should also document it.

Reply:
> Do you mean the if at lib/dma.c:407?
Yes.
> If yes then that line wont be reached in case of EACESS as the loop will break at L387
I'm referring to a scenario where the `if` is taken in L407 or L411 (the `while` loop won't have broken in L387). In this case we return `-1` but the contents of `errno` are undefined (who knows what syscall got executed earlier), we need to clear to `0`.

Reply:
Ok clearing errno, but I do not see anything which will set errno.

Reply:
`errno` might contain whatever the previous syscall did, e.g.
```
open(...); /* failed or some reason with errno == ENOENT */
dma_init_sg(...) /* fails because the region wasn't found, no errno still contains ENOENT */